### PR TITLE
Settings data, write and read, folder restructure

### PR DESCRIPTION
**Pull request by amrit-choudhary**

### CODE REVIEW

**Comment by amrit-choudhary** [comment]
also changed text color, and added a new file for text input boxes

File: src/game_screens/credits_screen.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        Screen.__init__(self, pygame, res, surface)
+        self.font = pygame.font.SysFont('cambria', 60)
+        self.font2 = pygame.font.SysFont('cambria', 30)
+        self.buttons['Vineet'] =        Button(pygame, res, surface, [20, 150, 300, 50], "Vineet")
+        self.buttons['Amrit'] =         Button(pygame, res, surface, [20, 220, 300, 50], "Amrit")
+        self.buttons['Venturillo'] =    Button(pygame, res, surface, [20, 290, 300, 50], "Venturillo")
+        self.buttons['Divyang'] =       Button(pygame, res, surface, [20, 360, 300, 50], "Divyang")
+        self.buttons['Azmal'] =         Button(pygame, res, surface, [20, 430, 300, 50], "Azmal")
+        self.buttons['Abhinandan'] =    Button(pygame, res, surface, [20, 500, 300, 50], "Abhinandan")
+        self.buttons['Gitter'] =        Button(pygame, res, surface, [20, 570, 300, 50], "Gitter")
```
Comment (vineetjc) [inline src/game_screens/credits_screen.py]: Gitter, unnecessary, let's just chuck

File: src/game_screens/credits_screen.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        self.surface.blit(textsurface, (20, 0))
+        self.surface.blit(textsurface2, (20, 100))
+              
+        for button in self.buttons:
+            self.buttons[button].draw()
+
+        mouseup_event = next((x for x in events if x.type == MOUSEBUTTONUP), None)
+
+        if mouseup_event != None:
+            if self.buttons['Vineet'].check_click(mouseup_event.pos):
```
Comment (vineetjc) [inline src/game_screens/credits_screen.py]: Interestingly, if you even scroll up or down on a button, it's counted as one of these events for some reason, and I think I have the idea why but I think this is a general issue with all our buttons, so I'm gonna flag this as an issue right away

File: src/ui/pygame.textinput.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
+"""
```
Comment (vineetjc) [inline src/ui/pygame.textinput.py]: Not gonna search deeply right now but where do we use this file at all?

File: src/game_screens/credits_screen.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+            if self.buttons['Azmal'].check_click(mouseup_event.pos):
+                webbrowser.open_new_tab('https://github.com/MD-AZMAL')
+                return Game_mode.CREDITS
+
+            if self.buttons['Abhinandan'].check_click(mouseup_event.pos):
+                webbrowser.open_new_tab('https://github.com/kainthcool')
+                return Game_mode.CREDITS
+
+            if self.buttons['Gitter'].check_click(mouseup_event.pos):
```
Comment (vineetjc) [inline src/game_screens/credits_screen.py]: Help me with this one, as more contributors come on in, what happens?

File: src/game_screens/credits_screen.py (hunk 2 above)
Comment (amrit-choudhary) [reply src/game_screens/credits_screen.py]: nice catch

File: src/ui/pygame.textinput.py (hunk 3 above)
Comment (amrit-choudhary) [reply src/ui/pygame.textinput.py]: this is for creating a text input box
will be used for:
entering player name
some settings

**Pull request merged**